src/Client.ts

```diff
@@ -0,0 +1,10 @@
+import { EventEmitter } from 'events';
+import Socket from './gateway/WebSocket';
+
+export const client = new EventEmitter();

```

Comment:
I know this is just temporary but I am not a big fan of exporting instances like this. Global class shouldn't be a standard

Reply:
Yes it's temporary. What do you suggest?

Reply:
Well, why are you even instantiating the EventEmitter?

Reply:
We can leave the client class for another time, for now we can do with just a basic websocket connection which will handle at least core inbound packets. We can link both together at a future time

Reply:
yes I figured.
Although what I was trying to express here is that it we shouldn't export an instance of client (so an instance of EventEmitter) but rather a class. And we let the user instantiate that himself.
So I guess we don't really care indeed, although I'd typically just not do that at all for now since it will most likely not be the final solution and could lead to weird behaviour in the way you code (typically, your socket function does not use client as a parameter but rather import it globally which I'd prefer to avoid)

Reply:
Should we remove the Client.ts file and just focus on the WebSocket for now?

Reply:
Yeah that would be great

Reply:
Agreed. Done